src/freertos_drivers/esp32/Esp32Gpio.hxx

```diff
@@ -566,7 +566,7 @@ template <class Defs> struct GpioInputPUPD : public GpioInputPin<Defs, true, tru
     {                                                                          \
         static const gpio_num_t PIN_NUM = (gpio_num_t)NUM;                     \
     public:                                                                    \
-        static const gpio_num_t pin()                                          \
+        static gpio_num_t pin()                                                \
```

Comment:
I'm curious on the removal of const here, was this due to a compilation error/warning?

Reply:
Yes the compiler issued a warning on this.
There is no use of returning an integral type by value as const. This will not force the caller to assign it to a const variable. Const only makes a difference for pointer and reference types, where it talks about the object being pointed *to*.